astropy/coordinates/transformations.py

```diff
@@ -1347,7 +1347,7 @@ def _combine_statics(self, transforms):
 
             if (isinstance(lasttrans, StaticMatrixTransform) and
                     isinstance(currtrans, StaticMatrixTransform)):
-                combinedmat = np.dot(lasttrans.matrix, currtrans.matrix)
+                combinedmat = np.dot(currtrans.matrix, lasttrans.matrix)
```

Comment:
while we're add it, could we replace `np.dot` with `matrix_multiply` here?  Future-proofing against stacks of static transforms (seems unlikely, but one never knows...).